subsys/net/lib/fota_download/Kconfig

```diff
@@ -8,6 +8,7 @@ menuconfig FOTA_DOWNLOAD
 	bool "FOTA Download"
 	depends on DOWNLOAD_CLIENT
 	depends on DFU_TARGET
+	depends on SYS_HASH_FUNC32
```

Comment:
Actually.. It looks like we should use `select` instead of `depends on`.

```
warning: FOTA_DOWNLOAD (defined at /jenkins_workspace/workspace/latest_sdk-
nrf_PR-11904/nrf/subsys/net/lib/fota_download/Kconfig:7) was assigned the value 'y' but got the
value 'n'. Check these unsatisfied dependencies: SYS_HASH_FUNC32 (=n). See
```

All build failures seem to relate to this one. The SYS_HASH_FUNC32 is not enabled in any of the build, so they will fail. Dependencies are not automatically selected.

Reply:
Ok, I changed it to "select"